static/js/topic_list_data.js

```diff
@@ -2,11 +2,33 @@ import * as hash_util from "./hash_util";
 import * as muting from "./muting";
 import * as narrow_state from "./narrow_state";
 import * as stream_topic_history from "./stream_topic_history";
+import * as topic_list from "./topic_list";
 import * as unread from "./unread";
 
 const max_topics = 5;
 const max_topics_with_unread = 8;
 
+function filter_topic_by_search(topic_names) {
```

Comment:
Is this function a near duplicate of an existing `stream_sort.js` function?  I think I'd be much happier if we were calling:

`util.filter_by_word_prefixe_match(search_term, topic_names)`

so maybe you can extract that function from `stream_sort.js` as having that name and location, and then just call it here?

Reply:
Yeah, actually it was same function modified a bit to work with topics. So in a preparatory commit I have made generic function `filter_by_word_prefix_match(items, search_term, item_to_text)` where `item_to_text` is a function which gives field of item to be compared.
To add more to why this parameter was needed, because this function could be used by callers which want to sort items of different type based on different fields, so having this parameter reduces a lot of code duplication from the caller.

Reply:
Yeah, this makes sense, thanks!